test/TypeDecoder/nominal_types.swift

```diff
@@ -153,6 +159,8 @@ extension STSContainer℠ where T == STSOuter {
 
 // DEMANGLE-TYPE: $s13nominal_types12STSContainerO8SubclassC11ExtraNestedCyAA8STSOuterV_Si_G
 // CHECK-TYPE: STSContainer<STSOuter>.Subclass<Int>.ExtraNested
+// DEMANGLE-TYPE: $s13nominal_types12STSContainerO8SubclassC16ExtraNestedWhereCyAA8STSOuterV_AEyAI_AIG_G
+// CHECK-TYPE: STSContainer<STSOuter>.Subclass<STSContainer<STSOuter>.Subclass<STSOuter>>.ExtraNestedWhere
```

Comment:
I would appreciate some advice on the proper way or utility to generate mangled names for types and declarations — doing this via `-emit-ir` and intuitive parsing is way too cumbersome to be the truth, especially for aliases that get stripped during silgen.

Reply:
I don't have a good solution. When adding new tests here I just build with `-emit-ir -g` and grep through the resulting debug info.